CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -782,117 +427,385 @@ public async Task<bool> RemoveWorkerManagerServerFromDefaultResourcePoolAsync()
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Enable Processing on Worker Server");
+			}
+		}
 
+		private async Task UpdateCategoriesOnWorkerForProcessingAsync(HttpClient httpClient, int workerServerId)
+		{
+			var updateCategories = new
+			{
+				workerArtifactId = workerServerId,
+				categories = new string[]
+				{
+					"NativeImaging",
+					"BasicImaging",
+					"SaveAsPDF",
+					"Processing"
+				}
+			};
+			string request = JsonConvert.SerializeObject(updateCategories);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.UpdateCategoriesOnWorkerEndpointUrl, request);
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Update Categories on Worker Server");
+			}
+		}
 
-			using (IResourceServerManager resourceServerManager = ServiceFactory.CreateProxy<IResourceServerManager>())
-			using (IResourcePoolManager resourcePoolManager = ServiceFactory.CreateProxy<IResourcePoolManager>())
+		private async Task<int> GetWorkerManagerServerTypeArtifactIdAsync(HttpClient httpClient)
+		{
+			int workerManagerServerTypeArtifactId = -1;
+			HttpResponseMessage serverTypeQueryResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourcePool.ResourceServerTypeQueryEndpointUrl, "");
+			if (!serverTypeQueryResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to query for Worker Manager Server Types");
+			}
+			string queryServerTypesResultString = await serverTypeQueryResponse.Content.ReadAsStringAsync();
+			dynamic queryServerTypesResult = JsonConvert.DeserializeObject<dynamic>(queryServerTypesResultString);
+			foreach (dynamic obj in queryServerTypesResult)
 			{
-				// Check for Default Resource Pool
-				ResourcePoolQueryResultSet resultPools = await resourcePoolManager.QueryAsync(queryPool);
+				if (obj["Name"].ToString() == "Worker Manager Server")
+				{
+					workerManagerServerTypeArtifactId = Convert.ToInt32(obj["ArtifactID"].ToString());
+					break;
+				}
+			}
+
+			return workerManagerServerTypeArtifactId;
+		}
 
-				Console.WriteLine($"{nameof(RemoveWorkerServerFromDefaultResourcePoolAsync)} - Checking if Default Resource Pool exists");
-				if (resultPools.Success && resultPools.TotalCount > 0)
+		private async Task<int> GetWorkerServerTypeArtifactIdAsync(HttpClient httpClient)
+		{
+			int workerServerTypeArtifactId = -1;
+			HttpResponseMessage serverTypeQueryResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourcePool.ResourceServerTypeQueryEndpointUrl, "");
+			if (!serverTypeQueryResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to query for Worker Server Types");

```

Comment:
Is there additional details in the HttpResponse regarding the actual error. We should probably put that in the Exception we are throwing which will be helpful when troubleshooting.

Reply:
Added failure message